File: HospitalDuty.Domain/Entities/Employee.cs
using System;
using System.Collections.Generic;
using HospitalDuty.Domain.Enums;

namespace HospitalDuty.Domain.Entities;

public class Employee
{
    public Guid Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    // public string PhoneNumber { get; set; } = string.Empty;
    // public string ProfileImage { get; set; } = string.Empty;
    // public Role Role { get; set; } = Role.Staff;

    public Guid? DepartmentId { get; set; }
    public Department? Department { get; set; }

    public Guid? HospitalId { get; set; }
    public Hospital? Hospital { get; set; }
    public ICollection<Shift> Shifts { get; set; } = new List<Shift>();

    // Identity ile birebir ilişki
    public string? ApplicationUserId { get; set; }  // string
    public ApplicationUser? ApplicationUser { get; set; }
}

File: client/src/types/employee.ts
export interface Employee {
  id: string;
  firstName: string;
  lastName: string;
  fullName: string;
  email: string;
  phoneNumber: string;
  roles?: string[];
  departmentId: string;
  hospitalId: string;
  department: string;
  hospitalName: string;
  profileImagePath?: string;
  profileImageUrl?: string;
  hospital?: {
    id: string;
    name: string;
    phone: string;
    district: string;
    city: string;
    address: string;
    email: string;
    website: string;
  };
  applicationUserId?: string;
}

export interface CreateEmployeeRequest {
  id?: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber?: string;
  roles?: string[];
  departmentId?: string;
  hospitalId?: string;
  applicationUserId?: string;
}

export interface UpdateEmployeeRequest {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  roles: string[];
  departmentId: string;
  hospitalId: string;
}

export const Role = {
  HospitalDirector: 'HospitalDirector',
  DepartmentManager: 'DepartmentManager',
  DepartmentLeader: 'DepartmentLeader',
  Doctor: 'Doctor',
  Nurse: 'Nurse',
  Staff: 'Staff',
  SystemAdmin: 'SystemAdmin'
} as const;

export type Role = typeof Role[keyof typeof Role];
